src/Schema/Directives/CacheDirective.php

```diff
@@ -87,7 +87,8 @@ function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
 
                 // We found a matching value in the cache, so we can just return early
                 // without actually running the query
-                if ($value = $cache->get($cacheKey)) {
+                $value = $cache->get($cacheKey);
+                if ($value !== null) {
```

Comment:
What if the value is `null`?

Reply:
IMO, It is a bad idea to cache raw `null`.  

In Laravel cache, `null` is considered as "non-existant" value. As mentioned in [laravel documentation](https://laravel.com/docs/8.x/cache#checking-for-item-existence).

> The `has` method [...] will also return false if the item exists but its value is null



If caching `null` result from resolver is wanted, we must encapsulate the value. Something like :
```php
$cache->put($cacheKey, ["value" => $result], Carbon::now()->addSeconds($maxAge));
```

And then :

```php
if ($cache->offsetExists($cacheKey)) {
  return $cache->get($cacheKey)["value"];
}
```

But this would include some BC, specially if you migrate lighthouse without clearing cache. So handling `null` (again, _if_ it is something wanted) possibly deserves a dedicated PR.

Reply:
Makes sense. Please add the Link and snippet as a code comment so this knowledge is kept.

Reply:
Done.